### ISSUE TITLE
64x32 Panel don't work

### ISSUE THREAD

**Issue by henri-ngl** [hzeller/rpi-rgb-led-matrix#1364]
Hi!

First of all thank you again for all your work. It's amazing. 

So i just bought some 64x32 Panel 
![64x32](https://user-images.githubusercontent.com/79509755/141997353-aff73578-51ee-4003-9b14-a29b2b44e0b4.JPG)
that i'm an raspberry pi 3 and an electrodragon hat 
![rasp3+hat](https://user-images.githubusercontent.com/79509755/141997572-27fc4d47-de7c-4ca9-85b4-97feddfaf379.jpg)
; all powered with 5V-60A power supply.

The problem is that nothing shows on the screen. When i use my old 64x32 every thing is OK but those new screen won't work. i tried the example of your library with different flags but still won't work. the screen remain off
![example](https://user-images.githubusercontent.com/79509755/141998195-9b68eac6-f7a2-4b29-8f19-74842988f23a.JPG)

It have the identical information as the old one  but none of them work. It's there a way to figure it?

Thank you

**Comment by greatballoflazers** [comment]
According to [this](https://lucasteske.dev/lyd6168) it is not supported. It is believed to be a rebranded MBI5153 or ICN2053. (This has been reported multiple times.) This is a PWM based panel, which is not supported by this project.

It is possible to use receiver card, like the one mentioned in the post, directly via NetCard. There are a few people which have gotten this to work. There is some code out there also, however it was reported by one of them that there could be a stability issue in some cases for the PWM panels. There is a work around, kind of. Sender cards also work well. If yours works as ICN2053 there is some code out there for it on the ESP32. 

No one has really sat down and overhauled this code base to support PWM based panels for the code base. A long time ago someone mentioned it would be possible. However most of their posts were removed. These store PWM bits differently than non-PWM panels do. They also have a different state machine for multiplexing.

These panels appear in the higher density, lower Px panels. They tend to be a little more expensive. Some of these have fancy features too. They are nice for larger displays but are more expensive for smaller displays. There are some non-PWM panels at lower density but getting higher quality is a little harder here. Receiver card can do better than Pi due to connector count, however that depends if it matters which is a topic in of itself.

Lot of the panels are modular. The ICs are usually fairly interchangeable. They make runs of different configurations. Sometimes it is helpful to ask which configuration they have in which density.

**Comment by DerekRead** [comment]
I have the same panel and can almost get it to display scrolling text. "hello world!" is almost legible but rows are missing.

I've tried a few different identical panels to try to eliminate the panel as the issue, they all suffer the exact same missing lines so most likely this is software not hardware.

I believe the missing lines are likely because this panel is probably 1:16 scan (?) which does not seem to be supported by this project (I have not found a setting for this).

I am running this from an rpi 4b, so found I had to use the param "--led-slowdown-gpio".
Brightness is set to 25 to reduce any possibility that this is an amp issue, even though I have a 4A supply which should be more than enough.

Reading the docs it looks like I should be using --led-panel-type=4 (for ABCD) but that appears much worse than when I have it set to 2.

These are the best settings I've found so far (also note I'm using the adafruit hat):
`python3 runtext.py --led-rows=32 --led-cols=64 --led-multiplexing=1 --led-row-addr-type=2 --led-gpio-mapping=adafruit-hat --led-brightness=25 --led-slowdown-gpio=4
`
![image](https://user-images.githubusercontent.com/19498041/216860096-5ba0e4ab-202d-4239-ba8e-04c2e9d26550.png)

**Comment by DerekRead** [comment]
I see now that characters are also individually mirrored and maybe flipped. More clues. Hmmm.

**Comment by DerekRead** [comment]
@henri-ngl: I guess this may not help you if your panel is completely blank but I thought I'd post just in case, given that I have the identical panel (from your rear photo), and I have seen settings with this software where the panel will appear blank.

tldr: using an rpi4b and the adafruit hat all I need is this (yes, I see you are using electrodragon):
`runtext.py --led-rows=32 --led-cols=64 --led-gpio-mapping=adafruit-hat --led-slowdown-gpio=4`

I was initially thrown off because I didn't use --led-slowdown-gpio=4
and was trying all other variations of --led-panel-type & --led-multiplexing & --led-row-addr-type I found these were all slightly or badly messed up (some were close) and in a few cases the screen **would be blank** <-- which sounds like your issue,  maybe?

Maybe this is useless info for you but maybe not.

I also pulled the current version of this matrix code before doing my last tests. There were quite a lot of changes since I last tested. That might have altered the behaviour as well.

![LEDPanel](https://user-images.githubusercontent.com/19498041/216874273-edf35ba5-24f1-4c61-a161-c2f27b89ec75.gif)

**Comment by anandrajgupta** [comment]
--led-multiplexing=0

**Comment by davemaster** [comment]
> --led-multiplexing=0

I think that value is the default. Try with 0-14 (top I think) values with ONE PANEL (power supply?)

**Comment by DerekRead** [comment]
Yes, 0 is the default. Thanks. For me it's working great now (single panel for this project). I had seen other people with other settings and that threw me way off track testing a bunch of settings in combinations, which wasn't necessary at all.

So, for this panel (see pic from OP but using adafruit hat) all I need are these settings for single panel: `runtext.py --led-rows=32 --led-cols=64 --led-gpio-mapping=adafruit-hat --led-slowdown-gpio=4`

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
